### ISSUE TITLE
Error when using generics and extending an interface or abstract class

### ISSUE THREAD

**Issue by before** [immutables/immutables#348]
When building the following code
```Java
interface AbstractValue<T> {

    @Value.Immutable
    static abstract class Concrete1<T> implements AbstractValue<T> {
        @Value.Parameter
        abstract T reference();
    }

    @Value.Immutable
    static abstract class Concrete2<T> implements AbstractValue<T> {
        @Value.Parameter
        abstract T reference();
    }
}
```
the generator fails with
```
ImmutableConcrete2.java:214: error: illegal start of expression
      return ImmutableConcrete2<T>.validate(new ImmutableConcrete2<T>(null, reference));
                                  ^
ImmutableConcrete1.java:214: error: illegal start of expression
      return ImmutableConcrete1<T>.validate(new ImmutableConcrete1<T>(null, reference));
                                  ^
2 errors
```

**Comment by elucash** [comment]
Thank you for reporting this! need to be fixed

**Comment by elucash** [comment]
Released in 2.2.2. Thank you!
